src/app/archive/archive_lib/processor.ml

```diff
@@ -51,6 +51,18 @@ module Public_key = struct
           public_key
 end
 
+module Token_owners = struct
+  (* hash table of token owners, updated for each block *)
+  let owner_tbl : Account_id.t Token_id.Table.t = Token_id.Table.create ()
```

Comment:
How would this work when the archive process restarts? Should we initialize the table with the tokens from the database?

Reply:
You're only adding tokens when there's a transaction in a block that mentions them. In that case, those tokens will be in the `tokens_used` list, and hence, in this table.

Reply:
> Actually, tokens section in zkapps integration test should be sufficient, right? since it exercises the replayer flow now

We have to add that, right?